Fix request params left unused

zahatikoff opened the pull request:
# What Does This PR Fix
This PR adresses the Issue #30, where the Tenor API calls were not passing any of the parameters provided in the function, which lead to the said requests failing.

## Related Issues
Fixes #30

zahatikoff committed:
fix: Fix request params left unused

hjönk-hjönk, i fixed the thing and then i also did a couple renames to
separate the already cached clause and the one when you actually make a
request. Hopefully it works now!

zahatikoff commented on src/util/tenor-utils.ts:
```diff
@@ -170,20 +170,19 @@ export class TenorSingleton {
         // eslint-disable-next-line no-magic-numbers
-        if (gifArray.length !== 0) {
-            return (gifArray[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)]);
+        if (cachedGifs?.length !== 0) {
```
changed to the optional property as asked

zahatikoff commented on src/util/tenor-utils.ts:
```diff
@@ -170,20 +170,19 @@ export class TenorSingleton {
          */
 
-        /*eslint-disable*/
+        /*eslint-disable camelcase*/
```
specified the exact features needed to be disabled

zahatikoff commented on src/util/tenor-utils.ts:
```diff
@@ -194,9 +193,10 @@ export class TenorSingleton {
+        /*eslint-enable camelcase*/
         try {
-            const gifObjects: TenorResponseObject[] = (await this.client.get<TenorResponse>("/search")).data.results;
+            const gifObjects: TenorResponseObject[] = (await this.client.get<TenorResponse>("/search",{params:requestParams})).data.results;
```
added back the necessary parameters field to the request

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -170,20 +170,19 @@ export class TenorSingleton {
         // eslint-disable-next-line no-magic-numbers
-        if (gifArray.length !== 0) {
-            return (gifArray[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)]);
+        if (cachedGifs!==null && cachedGifs?.length !== 0) {
```
`cachedGifs?.length` is enough here

zahatikoff committed:
fix: remove redundant checks

MirageAegis committed:
fix: Fix index out of range and env

Fix a bug that relates to a constraint of the Tenor API, where
the response length is limited to 50, not 100; and use the right
env for the request key

Also cleaned up getInstance and getGifs

MirageAegis committed:
fix: Make pat command send pats instead of hugs

MirageAegis commented on src/commands/action/pat.ts:
```diff
@@ -48,7 +48,7 @@ export const command: Command = {
         const patter: User = <User> ctx.member.user;
         const patted: User = <User> ctx.options.getUser("target");
-        const gif: string = await tenor.getGifs("anime+hug");
+        const gif: string = await tenor.getGifs("anime+headpat");
```
I fixed this overlooked mistake.

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
-    private static cache: Map<string, string[]>;
+    private readonly cache: Map<string, string[]>;
 
     private constructor() {
         this.client = axios.create({
             baseURL: "https://tenor.googleapis.com/v2",
         });
-        return this;
+        this.cache = new Map();
+
+        setInterval(() => {
+            this.cache.clear();
+        }, TWELVE_HOURS);
     }
 
     public static getInstance(): TenorSingleton {
         if (!this.instance) {
             this.instance = new TenorSingleton();
-            this.cache = new Map();
-            setInterval(() => {
-                this.cache.clear();
-            }, TWELVE_HOURS);
         }
+
         return this.instance;
     }
```
I also cleaned up the `getInstance` function by creating the interval in the constructor.

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
-                gifArray[i] = gifURL;
-            }
-            TenorSingleton.cache.set(topic, gifArray);
-            return (gifArray[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)]);
+            gifObjects = (await this.client.get<TenorResponse>("/search", {
+                params:requestParams
+            })).data.results;
         } catch {
-            throw new TenorError;
+            throw new TenorError();
         }
+
+        // All gif URLs from the gif objects
+        const gifUrls: string[] = gifObjects.map(g => g.media_formats.gif.url);
+
+        this.cache.set(topic, gifUrls);        
+        return gifUrls[Math.floor(Math.random() * DEFAULT_RESPONSE_LIMIT)];
```
Moved as many lines as possible out of the try block.

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
+            throw new TenorError();
         }
+
+        // All gif URLs from the gif objects
+        const gifUrls: string[] = gifObjects.map(g => g.media_formats.gif.url);
```
And condensed the array populating to a mapping function.

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -130,85 +130,87 @@ type tenorMediaObject = {
 
-const DEFAULT_RESPONSE_LIMIT = 100;
-const TWELVE_HOURS = 43_200_000;
+const DEFAULT_RESPONSE_LIMIT: number = 50;
```
The `DEFAULT_RESPONSE_LIMIT` was also changed to reflect the limit put by Tenor.

MirageAegis reviewed (approved):
The commands seem to work now.

Btw, I took the liberty to clean up some things.
